helpers/version-push/action.yml

```
@@ -26,17 +27,13 @@ runs:
       run: git clone --depth 1 https://github.com/home-assistant/version.git /tmp/version
 
     - shell: bash
-      id: version
       run: |
-        version=$(jq --raw-output '.["${{ inputs.key }}"]' /tmp/version/${{ inputs.channel }}.json)
-        echo "::set-output name=version::$version"
-
-    - shell: bash
-      run: |
-        sed -i "s|\"${{ inputs.key }}\": \"${{ steps.version.outputs.version }}\"|\"${{ inputs.key }}\": \"${{ inputs.version }}\"|g" /tmp/version/${{ inputs.channel }}.json
+        temp=$(mktemp)
+        jq '.${{ inputs.key }} |= "${{ inputs.version }}"' "/tmp/version/${{ inputs.channel }}.json" > "${temp}"
+        mv "${temp}" "/tmp/version/${{ inputs.channel }}.json"
 
     - shell: bash
       run: |
         cd /tmp/version
-        git commit -am "Bump ${{ inputs.key }} from ${{ steps.version.outputs.version }} to ${{ inputs.version }}"
+        git commit -am "Bump ${{ inputs.key }} to ${{ inputs.version }}"
```

```suggestion
        message="Bump ${{ inputs.key }} to ${{ inputs.version }}"
        git commit -am "${message/[]/}"
```
Maybe? to strip out the `[]` from the message

---

Yeah feels a bit ad-hoc. Came up with another input, `key-description`. With that we can use whatever describes the jq filter best...